andela/ah-backend-prime
# #164798292 Create , List and Update user profile

#### What does this PR do?
- Creates, update and list user profiles
#### Description of tasks to be completed.
- Creation of a profile which is done using Django signals each time a user is registered
- Updating a profile can only be done by the owner/creator of the profile
- Listing of all profiles: can be viewed by only authenticated users
#### How should this be manually tested?
- Fetch this branch by running the command git fetch origin ft-create-user-profile-164798292 in the terminal.
- Check into this branch by running git checkout ft-create-user-profile-164798292
- Activate your virtual environment by running source venv/bin/activate
- Delete the current database by running drop database DB_NAME; in the psql shell
- Create a new database by running create database DB_NAME; in the psql shell
- Update the .env file with the database name
- Source the .env file with the updated database name
- Migrate by running python manage.py migrate
- Start the application by running python manage.py runserver
- To test the register user API endpoint, Open postman, choose the POST method and paste this registration url http://localhost:8000/api/v1/users/register/. The user profile of the registered user is created automatically at this point.
- To get the created profile, run this `GET` method in postman `http://127.0.0.1:8000/api/v1/profiles/`
- To get a single user profile, run the `GET` method with the individuals username `http://127.0.0.1:8000/api/v1/profiles/jedisupreme` in postman
- To update profile, run the `PUT` method with your username and the suffix edit `http://127.0.0.1:8000/api/v1/profiles/jedisupreme/edit`
#### What are the relevant pivotal tracker stories?
[#164798292](https://www.pivotaltracker.com/story/show/164798292)
#### Screenshots
- Getting a list of user profiles after registeration
![image](https://user-images.githubusercontent.com/36891299/55743813-f286c680-5a3b-11e9-8c26-6f72d2615c15.png)
- Getting a single user profile
![image](https://user-images.githubusercontent.com/36891299/55743836-05010000-5a3c-11e9-9f32-521bb6a69358.png)
- Updating a user profile
![image](https://user-images.githubusercontent.com/36891299/55743881-2235ce80-5a3c-11e9-8bbe-9f7c60f73a71.png)

[Finishes #164798292]
